src/TestFramework/Coverage/XmlReport/JUnitTestCaseSorter.php

```diff
@@ -84,10 +88,10 @@ private function uniqueByTestFile(array $coverageTestCases): array
         $usedFileNames = [];
         $uniqueTests = [];
 
-        foreach ($coverageTestCases as $coverageTestCase) {
-            if (!in_array($coverageTestCase->testFilePath, $usedFileNames, true)) {
-                $uniqueTests[] = $coverageTestCase;
-                $usedFileNames[] = $coverageTestCase->testFilePath;
+        foreach ($coverageTestCases as $coverageLineData) {
+            if (!array_key_exists($coverageLineData->testFilePath, $usedFileNames)) {
```

Comment:
would `(string) $converageLineData->testFilePath` do the job instead of ingoring in phpstan config `'#array_key_exists expects int\|string, string\|null given#'`?

Reply:
I was very surprised to learn that `(string) $var` is... 7% slower. I did test this several times to reconfirm.

And yes, we know it is never null here. It's from JUnit report.

Reply:
interesting. Probably this will work faster

```php
assert($coverageLineData->testFilePath !== null);
```

Anyway, thanks for sharing. The current implementation is fine then

Reply:
`assert()` here is about as slow :man_shrugging:

Reply:
I see, thanks for checking!

Reply:
Shall we merge this?

For posterity, here's the benchmark I used.

<details>
  <summary>benchmark.php</summary>

```php
<?php
declare(strict_types=1);

use Infection\TestFramework\Coverage\XmlReport\JUnitTestCaseSorter;
use Infection\TestFramework\Coverage\XmlReport\JUnitTestCaseSorterArrayMap;
use Infection\TestFramework\Coverage\XmlReport\JUnitTestCaseSorterCast;
use Infection\TestFramework\Coverage\XmlReport\JUnitTestCaseSorterOld;

include 'vendor/autoload.php';

// To log, inside getUniqueSortedFileNames() add:
// file_put_contents('/tmp/JUnitTestCaseSorter.log', base64_encode(serialize($coverageTestCases)).PHP_EOL, FILE_APPEND | LOCK_EX);

$testCases = iterator_to_array((static function () {
    $file = fopen('/tmp/JUnitTestCaseSorter.log', 'r');

    while (($line = fgets($file)) !== false) {
        yield unserialize(base64_decode($line));
    }
    fclose($file);
})());

$sorters = [
    new JUnitTestCaseSorterArrayMap(),
    new JUnitTestCaseSorterCast(),
    new JUnitTestCaseSorter(),
    new JUnitTestCaseSorterOld(),
];

echo "Warm up... \n";

foreach ($sorters as $caseSorter) {
    foreach ($testCases as $coverageTestCases) {
        foreach ($caseSorter->getUniqueSortedFileNames($coverageTestCases) as $filename) {
        }
    }
}

echo "Testing...\n";

$results = [];

foreach ($sorters as $caseSorter) {
    $start = microtime(true);

    for ($i = 0; $i < 10; ++$i) {
        foreach ($testCases as $coverageTestCases) {
            foreach ($caseSorter->getUniqueSortedFileNames($coverageTestCases) as $filename) {
                $filename;
            }
        }
    }

    $total = microtime(true) - $start;

    $class = get_class($caseSorter);
    $results[$class] = $total;
    echo "$class: {$total}\n";
}

echo 'New vs Old: ' . round(100 * $results[JUnitTestCaseSorter::class] / $results[JUnitTestCaseSorterOld::class]) . "\n";
echo 'Iter vs Array: ' . round(100 * $results[JUnitTestCaseSorter::class] / $results[JUnitTestCaseSorterArrayMap::class]) . "\n";
echo 'Ignore vs Cast: ' . round(100 * $results[JUnitTestCaseSorter::class] / $results[JUnitTestCaseSorterCast::class]) . "\n";

```
</details>

Reply:
> Shall we merge this?

Sure, thought you can do it yourself since I removed PrettyCI integration completele. But now I see builds are still in progress.

Let's get it merged